File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/types/CognitoSignUpRequest.kt
package com.amazonaws.amplify.amplify_auth_cognito

data class CognitoSignUpRequest(val map: HashMap<String, *>) {
    val username: String? = map["username"] as String?;
    val password: String = map["password"] as String;
    val userAttributes: HashMap<String, String> = map["userAttributes"] as HashMap<String, String>;
    val providerOptions: HashMap<String, *>? = map["providerOptions"] as HashMap<String, *>?;
}

File: packages/amplify_auth_cognito/ios/Classes/CognitoSignUpRequest.swift
//
//  FlutterSignUpRequest.swift
//  amplify_auth
//
//  Created by Noyes, Dustin on 7/2/20.
//

import Foundation
struct CognitoSignUpRequest {
  var username: String?
  var password: String?
  var userAttributes: Dictionary<String, String> = [:]
  var providerOptions: Dictionary<String, Any>? = [:]
  init(dict: NSMutableDictionary){
    self.username = dict["username"] as! String?
    self.password = dict["password"] as! String?
    self.userAttributes = dict["userAttributes"] as! Dictionary<String, String>
    self.providerOptions = dict["providerOptions"] as! Dictionary<String, Any>?
  }
}
